doc/mutations/validations.md

```diff
@@ -0,0 +1,232 @@
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * CreateArticle constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param array $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   */
+  public function __construct(array $configuration, string $plugin_id, array $plugin_definition, AccountInterface $current_user) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * Creates an article.
+   *
+   * @param array $data
+   *   The title of the job.
+   *
+   * @return \Drupal\node\NodeInterface
+   *   The newly created article.
+   *
+   * @throws \Exception
+   */
+  public function resolve(array $data) {
+    $response = new ArticleResponse();
+    if ($this->currentUser->hasPermission("create article content")) {
+      $values = [
+          'title' => $data['title'],
+          'field_article_creator' => $data['creator'],
+      ];
+      $node = Node::create($values);
+      $node->save();
+      $response->setArticle($node);
+    }else {
+      $response->addViolation(
+      $this->t('You do not have permissions to create articles.')
+    );
+    }
+    return $response;
+  }
+
+}
+
+```
+
+We have added a new type that is returned `$response` where we call the `setArticle` method and if there are some validation errors we call the `addValidation` method to register in the errors property. Next we will resolve both these fields.
+
+## Resolve errors and article
+
+To resolve our fields similar to before we go to our schema implementation and call the created data producer `create_article` inside the `getResolverRegistry` method.
+
+```php
+/**
+ * {@inheritdoc}
+ */
+protected function getResolverRegistry() {
+
+  ...
+
+  // Resolve violations which are stored under "errors" key.
+  if ($source instanceof ResponseInterface && $fieldName == 'errors') {
+    return $source->getViolations();
+  }
+
+  // Create article mutation.
+  $registry->addFieldResolver('Mutation', 'createArticle',
+    $builder->produce('create_article')
+      ->map('data', $builder->fromArgument('data'))
+  );
+
+  // WHAT DOES THE USER NEED TO ADD HERE SO THAT
+  // 1 ) ArticleRespones is even usable
+  // 2 ) Resolve fields from it like the article and errors (maybe errors already works with code above)
+
+
+  ...
+  return $registry;
+}
```

Comment:
@joaogarin I checked GraphQL module, and you were right, that you should be able to register resolvers without having an schema extension plugin, just having the schema plugin. The final code (I believe which should work) would look like this:
```
<?php

declare(strict_types = 1);

namespace Drupal\example\Plugin\GraphQL\Schema;

use Drupal\graphql\GraphQL\Execution\ResolveContext;
use Drupal\graphql\GraphQL\ResolverBuilder;
use Drupal\graphql\GraphQL\ResolverRegistry;
use Drupal\graphql\GraphQL\Response\ResponseInterface;
use Drupal\graphql\Plugin\GraphQL\Schema\ComposableSchema;
use GraphQL\Type\Definition\ResolveInfo;

/**
 * Example GraphQL composable schema.
 *
 * @Schema(
 *   id = "example",
 *   name = "Example schema"
 * )
 */
class ExampleGraphqlSchema extends ComposableSchema {

  /**
   * {@inheritdoc}
   */
  public function getResolverRegistry() {
    // Initialize registry which will register all the field and type resolvers
    // we'd need. Initialize it with default field resolver which is used as
    // a fallback option for generic fields.
    $registry = new ResolverRegistry([
      __CLASS__,
      'defaultFieldResolver',
    ]);

    // Initialize builder which is used to build the resolving logic for the
    // fields. This includes the output data which is are going to be produced,
    // the inputs required for resolving them (resolver arguments, other static
    // or dynamic values, or even values produced by parent resolver), then the
    // contexts which the resolver can be aware of (eg language), and other
    // essentials.
    $builder = new ResolverBuilder();

    // Create an article mutation which produces custom ContentResponse
    // containing the article node.
    $registry->addFieldResolver('Mutation', 'createArticle',
      $builder->produce('create_article')
        ->map('data', $builder->fromArgument('data'))
    );

    return $registry;
  }

  /**
   * The default field resolver.
   *
   * Used if no field resolver was explicitly registered.
   *
   * @param array|\ArrayAccess|object $source
   *   The source (parent) value.
   * @param array $args
   *   An array of arguments.
   * @param \Drupal\graphql\GraphQL\Execution\ResolveContext $context
   *   The context object.
   * @param \GraphQL\Type\Definition\ResolveInfo $info
   *   The resolve info object.
   *
   * @return \Drupal\graphql\GraphQL\Resolver\ResolverInterface|\Closure|array
   *   The result for the field.
   */
  public static function defaultFieldResolver($source, array $args, ResolveContext $context, ResolveInfo $info) {
    $field_name = $info->fieldName;
    $property = NULL;

    if ($source instanceof ResponseInterface) {
      // Resolves violations which are defined under "error" key in the example
      // schema.
      if ($field_name == 'errors') {
        return $source->getViolations();
      }

      // Allow automatic resolving of fields on Response objects with the same
      // name as the method on that Response object. For example, this will
      // automatically resolves "article" field as there is a method on
      // ArticleResponse object "ArticleResponse::article()" which returns
      // article node.
      if (is_callable([$source, $field_name])) {
        $property = [$source, $field_name];
        return $property($source, $args, $context, $info);
      }
    }

    return $property;
  }

}
```